app/uk/gov/hmrc/incometaxsessiondata/services/UserDetailsService.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsessiondata.services
+
+import uk.gov.hmrc.auth.core.AffinityGroup.Individual
+import uk.gov.hmrc.incometaxsessiondata.models.{MtdItUser, Name, SuccessResponse}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.Future
+
+@Singleton
+class UserDetailsService @Inject(){
```

Comment:
Session instead of User
+ if we can be specific on the service contract/ returning type
-> Future[Either[Throwable, Unit]] - for create 
-> Future[Either[Throwable, Session]] - on get operation